gulp-tasks/utils/publish-helpers.js

```diff
@@ -0,0 +1,179 @@
+const fs = require('fs-extra');
+const path = require('path');
+const glob = require('glob');
+const archiver = require('archiver');
+const oneLine = require('common-tags').oneLine;
+
+const spawn = require('./spawn-promise-wrapper');
+const constants = require('./constants');
+const logHelper = require('../../infra/utils/log-helper');
+
+const SOURCE_CODE_DIR = 'source-code';
+const GROUPED_BUILD_FILES = 'grouped-build-files';
+
+const doesDirectoryExist = async (directoryPath) => {
+  let stats = null;
+  try {
+    stats = await fs.stat(directoryPath);
+  } catch (err) {
+    // NOOP
+  }
+  return stats !== null;
+};
+
+const getBuildPath = (tagName) => {
+  const tempReleasePath = path.join(
+    __dirname, '..', '..', constants.GENERATED_RELEASE_FILES_DIRNAME);
+  return path.join(tempReleasePath, tagName);
+};
+
+const downloadGitCommit = async (tagName, gitBranch) => {
+  if (!tagName) {
+    throw new Error(`You must provide a tagName to 'downloadGitCommit()`);
+  }
+
+  if (!gitBranch) {
+    throw new Error(`You must provide a gitBranch to 'downloadGitCommit()`);
+  }
+
+  const sourceCodePath = path.join(getBuildPath(tagName), SOURCE_CODE_DIR);
+
+  logHelper.log(oneLine`
+    Download Git Commit ${logHelper.highlight(gitBranch)}.
+  `);
+
+  const dirExists = await doesDirectoryExist(sourceCodePath);
```

Comment:
I'm a little fuzzy on the logic here—`doesDirectoryExist()` takes in `sourceCodePath`, and `sourceCodePath` uniquely depends on `tagName`. But it doesn't look like `gitBranch` is being checked.

So it seems like you'd skip `git clone` when the directory for the `tagName` exists, even if it might not have `gitBranch` checked out. Could you run into issues if the directory doesn't have the branch checked out?

Reply:
This directory should only ever be created by this code. The only scenario where what you've mentioned has occurred if someone / something created the directory in the expected location.